src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -146,9 +146,25 @@ private function parseDefinitions($content, $file)
         if (!is_array($content['services'])) {
             throw new InvalidArgumentException(sprintf('The "services" key should contain an array in %s. Check your YAML syntax.', $file));
         }
+        if (isset($content['services']['_defaults'])) {
```

Comment:
should we check whether the array contains extra keys likely to be used by a real service, and treat it as a service instead in this case (but with a deprecation warning) ? I would say that ``alias``, ``class`` and ``factory`` would cover it